File: decompiled/3.19.1/be/ehealth/businessconnector/therlink/domain/requests/GetTherapeuticLinkRequest.java
package be.ehealth.businessconnector.therlink.domain.requests;

import be.ehealth.businessconnector.therlink.domain.Author;
import be.ehealth.businessconnector.therlink.domain.Proof;
import be.ehealth.businessconnector.therlink.domain.TherapeuticLink;
import be.ehealth.businessconnector.therlink.domain.TherapeuticLinkRequestType;
import java.util.Date;
import org.joda.time.DateTime;

public class GetTherapeuticLinkRequest extends TherapeuticLinkRequestType {
   private static final long serialVersionUID = 1L;
   private int maxRows;

   public int getMaxRows() {
      return this.maxRows;
   }

   public void setMaxRows(int maxRows) {
      this.maxRows = maxRows;
   }

   /** @deprecated */
   @Deprecated
   public GetTherapeuticLinkRequest(String id, Date date, Author author, TherapeuticLink link, int maxRows, Proof... proofs) {
      super(id, date, author, link, proofs);
      this.maxRows = maxRows;
   }

   public GetTherapeuticLinkRequest(DateTime date, String id, Author author, TherapeuticLink link, int maxRows, Proof... proofs) {
      super(id, date, author, link, proofs);
      this.maxRows = maxRows;
   }
}

File: src/main/kotlin/org/taktik/connector/business/therlink/domain/requests/GetTherapeuticLinkRequest.kt
package org.taktik.connector.business.therlink.domain.requests

import org.taktik.connector.business.therlink.domain.Author
import org.taktik.connector.business.therlink.domain.Proof
import org.taktik.connector.business.therlink.domain.TherapeuticLink
import org.taktik.connector.business.therlink.domain.TherapeuticLinkRequestType
import java.util.Date
import org.joda.time.DateTime
import java.util.ArrayList

class GetTherapeuticLinkRequest(
    id: String,
    dateTime: DateTime,
    author: Author,
    link: TherapeuticLink? = null,
    proofs: List<Proof?> = ArrayList()
) : TherapeuticLinkRequestType(dateTime, id, author, link, proofs) {
    var maxRows: Int = 0

    @Deprecated("") constructor(
        id: String,
        date: Date,
        author: Author,
        link: TherapeuticLink?,
        maxRows: Int,
        vararg proofs: Proof?
    ) : this(id, DateTime(date), author, link, proofs.toList()) {
        this.maxRows = maxRows
    }

    constructor(
        date: DateTime,
        id: String,
        author: Author,
        link: TherapeuticLink?,
        maxRows: Int,
        vararg proofs: Proof?
    ) : this(id, date, author, link, proofs.toList()) {
        this.maxRows = maxRows
    }

    companion object {
        private val serialVersionUID = 1L
    }
}
